Shell/PredicateDefinition.cpp

```diff
@@ -899,6 +899,13 @@ void PredicateDefinition::count (TermList ts,int add, Unit* unit)
         count(TermList(sd->getTupleTerm()), add, unit);
         break;
 
+      case Term::SpecialFunctor::LAMBDA:
```

Comment:
I don't know what the function count i supposed to do .. maybe a comment or a more self-explanatory name would be good if you know/managed to figure out what it does. Buuuut intuitively I think it should probably call `count` on the lambda variable as well, no? I think the fall-through we had before was not the correct behaviour but noone ever added the missing case. If we don't know what this is supposed to do I'd rather add an unimplemented and not a fall through.

Reply:
OK - I've no idea, I was going for not breaking anything too much, but you're right - let's be told when we've not done it correctly. Can you replace these fallthroughs with `NOT_IMPLEMENTED` when you get a second please? Hacking on another branch atm.

Reply:
Disregard request, got to it now. :-)